### PR TITLE
2878 fix ci

### PR DESCRIPTION
**Pull request by rizzkhan1**
### Context
Build must terminate if failures are detected in `Run Tests` task 

### Changes proposed in this pull request
Put if condition to validate test results

### Checklist
https://trello.com/c/72HRuTa5/2878-figure-out-why-master-is-failing

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -38,6 +38,12 @@ steps:
     set -x
     docker run --name test-run-image -d $(IMAGE_NAME)
     docker exec test-run-image rake "parallel:spec[,, -O .azure_parallel]"
```
Comment (gpeng) [inline azure-pipelines.yml]: Should this be running default `rake` to include linting and brakeman?

File: azure-pipelines.yml [hunk 2]
```diff
@@ -38,6 +38,12 @@ steps:
...
+    test_result=$?
+      if [ "$test_result" != "0" ]
+      then
+          echo "##vso[task.logissue type=error]Cucumber test task exited abnormally."
```
Comment (gpeng) [inline azure-pipelines.yml]: I don't think this has Cucumber tests, they're rspec.

File: azure-pipelines.yml (hunk 1 above)
Comment (rizzkhan1) [reply azure-pipelines.yml]: Not sure tbh. There are separate tasks for linting and brakeman. @TobyRet setup the parallel testing for us so might be able to comment

Comment (gpeng) [reply azure-pipelines.yml]: If they're run elsewhere then that's fine. Just wasn't sure 👍

**Pull request merged**